### DISCUSSION TITLE
Clarification about relationship.foreign_key

### DISCUSSION THREAD

**Discussion by valhuber** [sqlalchemy/sqlalchemy#10034] [Usage Questions]
I understand [this description](https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.foreign_keys) in the context of multiple relationships, with *single field keys*.

I am unclear about *multi-field keys* -- it appears that the you simply provide a list of foreign key fields.  But, if there are multiple such fields, can the system match them to the parent primary key?  Or, does the system require a list of *field pairs*, such as `child.fk1=parent.pk1, child.fk2=parent.pk2`, etc

**Comment by zzzeek** [comment]
the "foreign_keys" parameter and/or "foreign()" annotation doesn't actually care about what each column in question is "foreign" against, only that this is a column that will be receiving the value of another column when the relationship is flushed.    The bigger question of *what* column gets populated is ultimately derived from the "primaryjoin" condition (or "secondaryjoin" if applicable).   if your tables make use of `ForeignKeyConstraint` to set up composite foreign keys, then the "primaryjoin" is derived from that.  if there is not a `ForeignKeyConstraint` present, then you would need to indicate a full "primaryjoin".

the "field pairs" that you refer towards do exist within the relationship configuration, but they are derived automatically from the combination of primaryjoin + set of columns in "primaryjoin" that are foreign.
